### ISSUE TITLE
clang static analyzer on modules

### ISSUE THREAD

**Issue by alfredh** [baresip/baresip#1368]
I tested with running clang analyzer on the modules.

it found some issues, that we should attempt to fix before the release.

the warnings are here:

```
modules/multicast/multicast.c:78:6: warning: Branch condition evaluates to a garbage value
        if (!codec) {
            ^~~~~~
1 warning generated.
modules/multicast/sender.c:149:2: warning: Value stored to 'mcsender' is never read
        mcsender = mem_deref(mcsender);
        ^          ~~~~~~~~~~~~~~~~~~~
modules/multicast/sender.c:192:3: warning: Value stored to 'mcsender' is never read
                mcsender = mem_deref(mcsender);
                ^          ~~~~~~~~~~~~~~~~~~~
2 warnings generated.
modules/multicast/receiver.c:477:2: warning: Value stored to 'mcreceiver' is never read
        mcreceiver = mem_deref(mcreceiver);
        ^            ~~~~~~~~~~~~~~~~~~~~~
modules/multicast/receiver.c:555:3: warning: Value stored to 'mcreceiver' is never read
                mcreceiver = mem_deref(mcreceiver);
                ^            ~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.
modules/multicast/player.c:444:2: warning: Value stored to 'err' is never read
        err |= str_dup(&player->device, cfg->play_dev);
        ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
modules/multicast/source.c:568:3: warning: Value stored to 'src' is never read
                src = mem_deref(src);
                ^     ~~~~~~~~~~~~~~
1 warning generated.
modules/aufile/aufile.c:167:4: warning: Value stored to 'mb' is never read
                        mb = mem_deref(mb);
                        ^    ~~~~~~~~~~~~~
modules/aufile/aufile.c:181:4: warning: Value stored to 'mb' is never read
                        mb = mem_deref(mb);
                        ^    ~~~~~~~~~~~~~
modules/aufile/aufile.c:186:4: warning: Value stored to 'err' is never read
                        err = ENOSYS;
                        ^     ~~~~~~
3 warnings generated.
modules/coreaudio/coreaudio.c:99:3: warning: Value stored to 'status' is never read
                status = AudioObjectGetPropertyDataSize(audioDevices[i],
                ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
modules/gtk/call_window.c:527:6: warning: Null pointer argument in call to string comparison function
        if (strncmp(reason, user_trigger_reason,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
modules/httpreq/httpreq.c:242:4: warning: Value stored to 'err' is never read
                        err = re_regex(carg->prm, strlen(carg->prm), "[^ ]*",
                        ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
modules/httpreq/httpreq.c:397:2: warning: Value stored to 'err' is never read
        err = http_client_add_ca(d->client, parm);
        ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
modules/httpreq/httpreq.c:506:2: warning: Value stored to 'err' is never read
        err |= conf_apply(conf_cur(), "httpreq_ca", ca_handler, d->client);
        ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 warnings generated.
modules/opus_multistream/opus_multistream.c:156:3: warning: Value stored to 'p' is never read
                p += n;
                ^    ~
1 warning generated.
```

**Comment by cspiel1** [comment]
Is it ok, if we fix this tomorrow?

**Comment by alfredh** [comment]
great! I have also fixed a couple of the warnings..

**Comment by alfredh** [comment]
thanks! after your fix the number of warnings are less:


```
modules/aufile/aufile.c:167:4: warning: Value stored to 'mb' is never read
                        mb = mem_deref(mb);
                        ^    ~~~~~~~~~~~~~
modules/aufile/aufile.c:181:4: warning: Value stored to 'mb' is never read
                        mb = mem_deref(mb);
                        ^    ~~~~~~~~~~~~~
modules/aufile/aufile.c:186:4: warning: Value stored to 'err' is never read
                        err = ENOSYS;
                        ^     ~~~~~~
3 warnings generated.
modules/avcodec/decode.c:229:2: warning: 'av_init_packet' is deprecated [-Wdeprecated-declarations]
        av_init_packet(&avpkt);
        ^
/opt/local/include/libavcodec/packet.h:487:1: note: 'av_init_packet' has been explicitly marked deprecated here
attribute_deprecated
^
/opt/local/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
1 warning generated.
modules/avformat/avformat.c:109:4: warning: 'av_init_packet' is deprecated [-Wdeprecated-declarations]
                        av_init_packet(&pkt);
                        ^
/opt/local/include/libavcodec/packet.h:487:1: note: 'av_init_packet' has been explicitly marked deprecated here
attribute_deprecated
^
/opt/local/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
1 warning generated.
modules/gtk/call_window.c:527:6: warning: Null pointer argument in call to string comparison function
        if (strncmp(reason, user_trigger_reason,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
modules/httpreq/httpreq.c:245:4: warning: Value stored to 'err' is never read
                        err = re_regex(carg->prm, strlen(carg->prm), "[^ ]*",
                        ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
modules/httpreq/httpreq.c:400:2: warning: Value stored to 'err' is never read
        err = http_client_add_ca(d->client, parm);
        ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
modules/httpreq/httpreq.c:509:2: warning: Value stored to 'err' is never read
        err |= conf_apply(conf_cur(), "httpreq_ca", ca_handler, d->client);
        ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 warnings generated.
```

**Comment by cspiel1** [comment]
Will fix aufile and httpreq warnings.

**Comment by alfredh** [comment]
thanks for helping to fix the warnings!

all the warnings should be fixed now in commit e10e22e3c8126d6f0e345dd3062b5e84cac9a443
